### PR TITLE
Recommend `init` keyword

### PR DESCRIPTION
**Pull request by jcouv**

### CODE REVIEW

**Comment by jcouv** [comment]
Tagging @CyrusNajmabadi for small review. Trying to get this in before Monday so it's in the bits Mads demos.

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/InitKeywordRecommender.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        public InitKeywordRecommender()
+            : base(SyntaxKind.InitKeyword)
+        {
+        }
+
+        protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
+        {
+            return
+                context.TargetToken.IsAccessorDeclarationContext<PropertyDeclarationSyntax>(position, SyntaxKind.InitKeyword) ||
+                context.TargetToken.IsAccessorDeclarationContext<IndexerDeclarationSyntax>(position, SyntaxKind.InitKeyword);
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/InitKeywordRecommender.cs:22]: don't we need:

`c#
class C
{
    init 
    {
    }
}
```

as well?

File: src/Features/CSharp/Portable/Completion/CompletionProviders/KeywordCompletionProvider.cs [hunk 2]
```diff
@@ -138,6 +138,7 @@ private static ImmutableArray<IKeywordRecommender<CSharpSyntaxContext>> GetKeywo
                 new SealedKeywordRecommender(),
                 new SelectKeywordRecommender(),
                 new SetKeywordRecommender(),
+                new InitKeywordRecommender(),
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/KeywordCompletionProvider.cs:141]: sort.

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/InitKeywordRecommender.cs (hunk 1 above)
Comment (jcouv) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/InitKeywordRecommender.cs:22]: No, we brainstormed that, but that wasn't decided yet.

**Pull request merged**